chain/sub/ratelimit/queue_test.go

```diff
@@ -0,0 +1,61 @@
+package ratelimit
+
+import (
+	"testing"
+)
+
+func TestQueue(t *testing.T) {
```

Comment:
No objection to this test, obviously, but I'm surprised it was needed to make the queue lok "used" -- it should already look used because of its use in rate limiting.

Do you understand why this was the case?

Reply:
Yes! I sort of understand, for some reason pop is unused:

```
chain/sub/ratelimit/queue.go:38:17: func `(*queue).pop` is unused (unused)
func (q *queue) pop() int64 {
```

it's pretty common to write a little datastructure library, write out a bunch of api/funcs for it, and then not use a few when you have your final implementation. I think there's some combination of peeking and truncating tha thappens in ./chain/sub/ratelimit/window.go instead, but i haven't done a thorough dive into it. Given my expectaiton of this pattern of building a general tool and then just using some of it, the test made more sense than deleting pop in case in future we wanted to reorganize this code

Reply:
Got it, thanks for the explanation! Makes total sense.